src/FuncSharp.Examples/ITry.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace FuncSharp.Examples
+{
+    public class ITryUsages
+    {
+        public class Person
+        {
+            public Person(string name, int age)
+            {
+                Name = name;
+                Age = age;
+            }
+
+            public string Name { get; }
+            public int Age { get; }
+        }
+
+        public enum PersonParsingError
+        {
+            NameNotProvided,
+            AgeNotANumber,
+            AgeNegative,
+            AgeTooHigh
+        }
+
+        public enum NetworkOperationError
+        {
+            ServerError,
+            ConnectionTimedOut,
+            NetworkIssues
+        }
+
+        public ITry<T, NetworkOperationError> PerformNetworkOperation<T>(Func<Unit, T> operation)
```

Comment:
For the purpose of the example, which is now pretty big what about splitting it two parts:
- One will be a class representing the API. It will define the NetworkOperationError, PerformNetworkOperation, DownloadNumber and TransformNumber. There we can showcase how to handled expected exceptions in a type safe manner. That's the simpler example for start and we use this pattern a lot.
- Second example would be the "business" layer, i.e. this file with Person, PersonParsingError and all the methods that use the API and work with its results. I think that would nicely correspond to how we actually do it in real life.

Reply:
That will also simplify the "business" layer examples since instead of `PerformNetworkOperation(_ => DownloadNumberOverNetwork())`, you'll just have `api.GetNumber()` and `api.Double(i)`.